pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -376,6 +464,53 @@ func (b *Base) Objects(rsrc interface{}, rsrclabels map[string]string, observed,
 		}
 	}
 
+	item := k8s.GetItem(observed, &batchv1.Job{}, getUpgradeJobName(master), "default")
```

Comment:
Why always fetch the Job? Shouldn't we check the status to see if there is upgrade in progress before getting the job?